File: payloads/go/inode_bomb.go
package main

import (
	"fmt"
	"os"
)

func main() {
	for i := 0; i < 100000; i++ {
		name := fmt.Sprintf("/tmp/inode_%d", i)
		err := os.WriteFile(name, []byte{}, 0644)
		if err != nil {
			panic(fmt.Sprintf("stopped at %d: %v", i, err))
		}
	}
	fmt.Println("ESCAPED")
}

File: payloads/javascript/inode_bomb.js
const fs = require("fs");

for (let i = 0; i < 100000; i++) {
  try {
    fs.writeFileSync("/tmp/inode_" + i, "x");
  } catch (e) {
    console.log("inode bomb stopped at i=" + i + ":", e.message);
    break;
  }
}
console.log("created 100000 files in /tmp");
